# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/SharedSurfacesParent.h
# Commit: 923dee1aab01
# Full Hash: 923dee1aab0128ea4ff60b876a8626af322084d8
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-15 05:00:55
# Description:
#   Bug 1869933 - Avoid allocating external image ID until ready for sharing with compositor process. r=gfx-reviewers,lsalzman
#   
#   There are some error paths where we could allocate an external image ID
#   for a SourceSurfaceSharedData before we are able to finish sharing it.
#   This patch makes it so that we defer allocating the ID until the last
# ==============================================================================

diff -r ef864a709ce1 -r 923dee1aab01 gfx/layers/ipc/SharedSurfacesParent.h
--- a/gfx/layers/ipc/SharedSurfacesParent.h	Thu Dec 14 23:17:37 2023 +0200
+++ b/gfx/layers/ipc/SharedSurfacesParent.h	Thu Dec 14 20:08:59 2023 +0000
@@ -28,7 +28,7 @@
 
 namespace layers {
 
-class SharedSurfacesChild;
+class CompositorManagerParent;
 class SharedSurfacesMemoryReport;
 
 class SharedSurfacesParent final {
@@ -69,7 +69,7 @@
   static bool AgeAndExpireOneGeneration();
 
  private:
-  friend class SharedSurfacesChild;
+  friend class CompositorManagerParent;
   friend class gfx::SourceSurfaceSharedDataWrapper;
 
   SharedSurfacesParent();
